core/backend/src/annotations/LeaderGeometry.ts

```diff
@@ -0,0 +1,183 @@
+import { ColorDef, ElementGeometry, GeometryParams, TextAnnotationLeader, TextFrameStyleProps } from "@itwin/core-common";
+import { CurveCurve, LineSegment3d, LineString3d, Loop, Path, Point3d, Transform, Vector3d } from "@itwin/core-geometry";
+import { computeFrame } from "./FrameGeometry";
+import { TextBlockLayout } from "./TextBlockLayout";
+
+/**
+ * Constructs and appends leader lines and their terminators to the provided geometry builder for a text annotation.
+ *
+ * This function processes an array of `TextAnnotationLeader` objects, computes their attachment points
+ * relative to a text frame (or a default rectangular frame if none is provided), and appends the leader
+ * line and terminator geometry to the builder. It also applies color overrides if specified
+ * in the leader's style overrides.
+ *
+ * @param builder - The geometry builder to which the leader geometries will be appended.
+ * @param leaders - An array of leader properties.
+ * @param layout - The layout information for the text block, including its range.
+ * @param transform - The transform to apply to the frame and leader geometry.
+ * @param params - The geometry parameters, such as color, to use for the leader lines.
+ * @param frame - (Optional) The style properties for the text frame. If not provided or set to "none", a default rectangle is used.
+ * @returns `true` if at least one leader with a terminator was successfully appended; otherwise, `false`.
+ * @beta
+ */
+export function appendLeadersToBuilder(builder: ElementGeometry.Builder, leaders: TextAnnotationLeader[], layout: TextBlockLayout, transform: Transform, params: GeometryParams, frame?: TextFrameStyleProps,): boolean {
+  // If there is no frame, use a rectangular frame to compute the attachmentPoints for leaders.
+  if (frame === undefined || frame.shape === "none") {
+    frame = { shape: "rectangle" }
+  }
+  if (frame.shape === undefined || frame.shape === "none") return false;
+  const frameCurve = computeFrame({ frame: frame.shape, range: layout.range, transform });
```

Comment:
I would pass this in to avoid duplicate calls to `computeFrame`. it would also remove a dependency to another file which could be good.